### ISSUE TITLE
N806 false-positive with a global variable

### ISSUE THREAD

**Issue by actionless** [charliermarsh/ruff#799]
```python
GLOBAL_VAR: str = 'foo'

def get_all_aur_names() -> str:
    global GLOBAL_VAR
    if not GLOBAL_VAR:
        GLOBAL_VAR = 'bar'
    return GLOBAL_VAR
```

```console
$ ruff ruff_global_N806.py
Found 1 error(s).
ruff_global_N806.py:6:9: N806 Variable `GLOBAL_VAR` in function should be lowercase
```

**Comment by charliermarsh** [comment]
Good catch, will fix.

**Comment by Borda** [comment]
@charliermarsh I see the same issue with `v0.0.233`
adding it in: https://github.com/Lightning-AI/metrics/pull/1464
```py
MAX_PORT = 8100
START_PORT = 8088
CURRENT_PORT = START_PORT


def setup_comm(rank, world_size):
    global CURRENT_PORT

    CURRENT_PORT += 1
    if CURRENT_PORT > MAX_PORT:
        CURRENT_PORT = START_PORT  # <- N806 Variable `CURRENT_PORT` in function should be lowercase
```
